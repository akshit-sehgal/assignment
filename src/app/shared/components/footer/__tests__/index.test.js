import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../';

describe('Testing <Footer/>', () => {
    it('should render the default view', () => {
        const wrapper = shallow(<Footer />);
        expect(wrapper).toMatchSnapshot();
        const lists = wrapper.find('.footer__list');
        expect(lists.length).toEqual(4);
        expect(lists.at(0).find('.footer__list-item').length).toEqual(3);
        expect(lists.at(1).find('.footer__list-item').length).toEqual(4);
        expect(lists.at(2).find('.footer__list-item').length).toEqual(4);
        expect(lists.at(3).find('.footer__list-item').length).toEqual(3);
    });
});