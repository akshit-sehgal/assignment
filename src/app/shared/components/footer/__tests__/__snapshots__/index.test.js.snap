// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Footer/> should render the default view 1`] = `
<div
  className="footer-container"
>
  <div
    className="footer__list-container section-container"
  >
    <div
      className="footer__list"
      key="About"
    >
      <div
        className="footer__list-head"
      >
        About
      </div>
      <button
        className="footer__list-item"
        key="Contact Us"
      >
        Contact Us
      </button>
      <button
        className="footer__list-item"
        key="About Us"
      >
        About Us
      </button>
      <button
        className="footer__list-item"
        key="Careers"
      >
        Careers
      </button>
    </div>
    <div
      className="footer__list"
      key="Help"
    >
      <div
        className="footer__list-head"
      >
        Help
      </div>
      <button
        className="footer__list-item"
        key="Payments"
      >
        Payments
      </button>
      <button
        className="footer__list-item"
        key="Shipping"
      >
        Shipping
      </button>
      <button
        className="footer__list-item"
        key="Cancellation & Returns"
      >
        Cancellation & Returns
      </button>
      <button
        className="footer__list-item"
        key="FAQs"
      >
        FAQs
      </button>
    </div>
    <div
      className="footer__list"
      key="Policy"
    >
      <div
        className="footer__list-head"
      >
        Policy
      </div>
      <button
        className="footer__list-item"
        key="Return Policy"
      >
        Return Policy
      </button>
      <button
        className="footer__list-item"
        key="Terms of Use"
      >
        Terms of Use
      </button>
      <button
        className="footer__list-item"
        key="Security"
      >
        Security
      </button>
      <button
        className="footer__list-item"
        key="Privacy"
      >
        Privacy
      </button>
    </div>
    <div
      className="footer__list"
      key="Social"
    >
      <div
        className="footer__list-head"
      >
        Social
      </div>
      <button
        className="footer__list-item"
        key="Twitter"
      >
        Twitter
      </button>
      <button
        className="footer__list-item"
        key="Facebook"
      >
        Facebook
      </button>
      <button
        className="footer__list-item"
        key="YouTube"
      >
        YouTube
      </button>
    </div>
  </div>
  <hr />
  <div
    className="footer__cr"
  >
    Tokopedia. All Rights Reserved Â© 2020
  </div>
</div>
`;
